src/pages/Account/Tabs/ModulesTab.tsx

```diff
@@ -179,6 +198,230 @@ function ViewCode({address}: {address: string}): JSX.Element {
+      <Box
+        sx={{
+          maxHeight: "500px",
+          overflowY: "auto",
+        }}
+      >
+        {Object.entries(moduleAndFnsGroup).map(([moduleName, fns]) => (
+          <Box key={moduleName} marginY={3}>
+            <Typography fontSize={14} fontWeight={500} marginBottom={"8px"}>
+              {moduleName}
+            </Typography>
+            {fns.map((fn) => {
+              const selected =
+                moduleName === selectedModuleName && fn.name === selectedFnName;
+              return (
+                <Box
+                  key={fn.name}
+                  onClick={() => handleClick(moduleName, fn.name)}
+                  fontSize={12}
+                  fontWeight={400}
+                  marginBottom={"8px"}
+                  padding={1}
+                  sx={{
+                    bgcolor: !selected
+                      ? "transparent"
+                      : theme.palette.mode === "dark"
+                      ? grey[500]
+                      : grey[200],
+                    ":hover": {
+                      cursor: "pointer",
+                    },
+                  }}
+                >
+                  {fn.name}
+                </Box>
+              );
+            })}
+          </Box>
+        ))}
+      </Box>
+    </Box>
+  );
+}
+
+function WriteContractForm({
+  module,
+  fn,
+}: {
+  module: Types.MoveModule;
+  fn: Types.MoveFunction;
+}) {
+  const [state] = useGlobalState();
+  const {account, connected} = useWallet();
+  const {handleSubmit, control} = useForm<WriteContractFormType>();
+  const {submitTransaction} = useSubmitTransaction();
+
+  const onSubmit: SubmitHandler<WriteContractFormType> = async (data) => {
+    const payload: Types.TransactionPayload = {
+      type: "entry_function_payload",
+      function: `${module.address}::${module.name}::${fn.name}`,
+      type_arguments: data.typeArgs,
+      arguments: data.args,
+    };
+    await submitTransaction(payload);
+  };
+
+  const hasSigner = fn.params.length > 0 && fn.params[0] === "&signer";
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Stack spacing={4}>
+        <Typography fontSize={14}>
+          {fn.name}
+          {fn.generic_type_params.length > 0 &&
+            "<" +
+              [...Array(fn.generic_type_params.length)].map((_, i) => `T${i}`) +
+              ">"}
+          ({fn.params.join(", ")})
+        </Typography>
+        <Stack spacing={4}>
+          {[...Array(fn.generic_type_params.length)].map((_, i) => (
+            <Controller
+              key={i}
+              name={`typeArgs.${i}`}
+              control={control}
+              render={({field: {onChange, value}}) => (
+                <TextField
+                  onChange={onChange}
+                  value={value}
+                  label={`T${i}`}
+                  fullWidth
+                />
+              )}
+            />
+          ))}
+          {fn.params.map((param, i) => {
```

Comment:
https://user-images.githubusercontent.com/121921928/224185683-c04cd52f-be1e-4eb0-8719-a316179f8fff.mov

seems like i can't perform the write?

Reply:
this actually prompts me to think: for any kinds of contract, "write" might not be that helpful if we don't have proper "read". in this delegation example, if i wanna `unlock`, i need to know how much i have in there for me to unlock right? otherwise, it's gonna be tab switching, searching and digging to figure out the number first, and then perform the write later. couldn't figure out an ideal way to handle this, cuz there's no very systematic way to connect btw "unlock" and "active funds you have". 
But one thing that might be helpful here is: when tab switching, we should still keep the value in the input. so that if someone just wants to go to "Read" to checkout their amount, they can do so without repetitively entering their address again and again. @0xbe1 

cc @0xZihan

Reply:
prefer storing the info in URL - can we worry about this routing change later? @jillxuu